### PR TITLE
[#618] Two-layer color system

### LINKED ISSUE #618: two-layer color system
Right now we generate a large number of colors, and make them accessible to the entire project using the percentage-values of the color mix as the names, via `palette.get()`. It’d be great to have a limited palette of menaingfully-named colors available in the project, selected from the large number of generated (but adjustable & overridable) variants.

- specify a map of base colors
- generate a palette from each, mixing white and black at specified percentages
- select colors from those base colors, use them in a semantically-named map of colors e.g.

```scss

"text": (
  "default": map.get(map.get($base-colors, "black"), "100"),
  "light": map.get(map.get($base-colors, "black"), "80"),
  "lighter": map.get(map.get($base-colors, "black"), "70"),
),
"background": (
  "default": map.get(map.get($base-colors, "white"), "100"),
),
"brand": (
  "default",
  "light",
  "active"
  "hover"
),
"border": (
// etc
)
```

Perhaps those maps can be used as default settings for the respective utility classes e.g. `u-border` uses "border", `u-fg` uses "text"

### PR DESCRIPTION
**Pull request by planktonic**
Fixes #618 

## Changes

- Adds three layers: base colors & setup, base palette, and semantic palette
- color generator now outputs shades and tints in one, no `-shades` palettes any more
- Updated storybook documentation & color palette presentation

## Summary

Color design tokens are now a 2.5-tier system:

- First tier is split into two: a short list of **base colors**, and large **base palettes** generated from them. These are both optional.
- The second tier is the **semantic color palettes**, selected from the base palettes. Those are a small selection of colors which will actually be used throughout the codebase, and are mostly named based on their usage (“background”, “text”, “brand-1”…)

## Looks like

screencapture-localhost-6006-iframe-html-2022-08-09-17_04_00

## Checks

Delete if not applicable:

- [x] Storybook documentation has been updated
- [x] Fixtures in [`test/scss/`](../test/scss/) have been updated
- [x] Your changes have been added to the `unreleased` section of [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

**Review by ArinzeJeffrey-droid — APPROVED** [review]
I left some comments but it looks good. ✨

File: scss/bitstyles/atoms/button--danger/_settings.scss [hunk 1]
```diff
@@ -5,33 +5,33 @@
 //
 // Hover colors ////////////////////////////////////////
 
-$color-hover: palette.get('white') !default;
-$background-color-hover: palette.get('danger', '100') !default;
-$border-color-hover: palette.get('danger', '100') !default;
+$color-hover: palette.get('grayscale', 'white') !default;
+$background-color-hover: palette.get('danger', 'base') !default;
+$border-color-hover: palette.get('danger', 'base') !default;
```
[the remark this answers is not in the document]
Comment (planktonic) [reply scss/bitstyles/atoms/button--danger/_settings.scss:10]: Yeh the buttons will be a bit broken in this branch — the follow-up PR is focused on buttons, and re-specifies all the button colors :) So for this PR, it’s OK

File: scss/bitstyles/atoms/button--icon-reversed/_settings.scss [hunk 2]
```diff
@@ -19,23 +19,26 @@ $box-shadow: none !default;
...
 
-$color-disabled: palette.get('black', '50') !default;
-$background-color-disabled: transparent !default;
-$border-color-disabled: transparent !default;
+$color-disabled: palette.get('text', 'lighter') !default;
+$background-color-disabled: rgba(
+  palette.get('background', 'dark'),
+  50%
+) !default;
+$border-color-disabled: palette.get('text', 'base') !default;
```
[the remark this answers is not in the document]
Comment (planktonic) [reply scss/bitstyles/atoms/button--icon-reversed/_settings.scss:43]: Thanks for checking these out! 💕 They’ll be fixed in #670

File: scss/bitstyles/atoms/button--mode/_settings.scss [hunk 3]
```diff
@@ -18,54 +18,54 @@ $icon-spacing: size.get('s') !default;
...
-$border-color-hover: palette.get('white', '95') !default;
+$color-hover: palette.get('text', 'base') !default;
+$background-color-hover: palette.get('text', 'lighter') !default;
+$border-color-hover: palette.get('text', 'lighter') !default;
 $box-shadow-hover: shadow.to-box-shadow(
   shadow.generate(
-    $color: rgba(palette.get('brand-1', '100'), 0.05),
-    $blur-radius: size.get('m'),
+    $color: rgba(palette.get('brand-1', 'base'), 0.025),
+    $blur-radius: size.get('l'),
```
[the remark this answers is not in the document]
Comment (planktonic) [reply scss/bitstyles/atoms/button--mode/_settings.scss:35]: Yeh that might be the way we go with buttons, it’s still in discussion :) Once #670 is merged too, let’s see how we feel about it

File: scss/bitstyles/settings/color-palette.stories.mdx [hunk 4]
```diff
@@ -89,64 +110,103 @@ The color variants are generated from the set of initial base colors. Changing t
...
 ```
 
-### Overriding the color-palette
+## Semantic Color palette
 
-For more fine-grained and total control, override the `$color-palette` Sass map. Note that you cannot add to nor edit the values in the default map. You can only overwrite the entire thing, so be sure to copy the colors you’ll use even if you don’t change them.
+All the colors of the base palette are available to you to construct the semantic palette. If the default colors are not suitable, or if changes to the base colors or base color palette mean the colors used by default in the semantic palette are no longer suitable, override the semantic palette. You must overwrite the entire thing, so be sure to copy the colors you’ll use even if you don’t change them.
 
-Depending on your requirements, you may want to use the color generator from bitstyles to generate some color variants, and manually specify others:
+It’s recomended that you select the colors from the base palette, but it’s not required:
```
[the remark this answers is not in the document]
Comment (planktonic) [reply scss/bitstyles/settings/color-palette.stories.mdx:191]: 💚

**Pull request merged**